contracts/margin/external/oracle/MakerOracle.sol

```diff
@@ -167,35 +144,36 @@ contract MakerOracle is
         onlyMargin
         returns (bool)
     {
-        require(this.positionUsesSupportedTokens(positionId));
         Margin margin = Margin(DYDX_MARGIN);
 
+        address heldToken = margin.getPositionHeldToken(positionId);
+        address owedToken = margin.getPositionOwedToken(positionId);
+
+        // Currently we only support ETH/DAI and DAI/ETH.
+        require(
+            (heldToken == DAI && owedToken == WETH) ||
+            (heldToken == WETH && owedToken == DAI)
+        );
+
         // The MakerDAO medianizer contract returns ETH/USD with 18 digits after
         // the decimal point.
         bytes32 oracleRead = IMedianizer(MAKERDAO_MEDIANIZER).read();
         uint256 ethUsdRate = uint256(oracleRead);
 
-        address heldToken = margin.getPositionHeldToken(positionId);
-        address owedToken = margin.getPositionOwedToken(positionId);
         uint256 heldAmount = margin.getPositionBalance(positionId);
         uint256 owedAmount = margin.getPositionOwedAmount(positionId);
 
         if (heldToken == DAI && owedToken == WETH) {
             // I.e. short position against ETH.
             return (
-                heldAmount * (10**8) * (10**18) >=
-                owedAmount * COLLATERAL_REQUIREMENT * ethUsdRate
-            );
-        } else if (heldToken == WETH && owedToken == DAI) {
-            // I.e. long position in ETH.
-            return (
-                heldAmount * (10**8) * ethUsdRate >=
-                owedAmount * COLLATERAL_REQUIREMENT * (10**18)
+                heldAmount.mul(10**26) >=
```

Comment:
It's a little more unclear with  the `10**26` and the `10**10`. Can you put in a comment explaining what you're doing here, or change it to `.mul((10**8) * (10**18))`

Reply:
Added comments